src/checker/nlprules.rs

```diff
@@ -0,0 +1,160 @@
+//! A nlp based rule checker base on `nlprule`
+//!
+//! Does check grammar, and is supposed to only check for grammar.
+//! Sentence splitting is done in hand-waving way. To be improved.
+
+use super::{Checker, Detector, Documentation, Suggestion, SuggestionSet};
+use crate::{CheckableChunk, ContentOrigin};
+
+use anyhow::Result;
+use fs_err as fs;
+use log::{debug, info, trace, warn};
+use rayon::prelude::*;
+
+use nlprule::types::Suggestion as NlpFix;
+use nlprule::{Rules, Tokenizer};
+
+static DEFAULT_TOKENIZER_BYTES: &[u8] =
+    include_bytes!(concat!(env!("OUT_DIR"), "/en_tokenizer.bin"));
+static DEFAULT_RULES_BYTES: &[u8] = include_bytes!(concat!(env!("OUT_DIR"), "/en_rules.bin"));
+
+pub(crate) struct NlpRulesChecker;
+
+impl Checker for NlpRulesChecker {
+    type Config = crate::config::NlpRulesConfig;
+
+    fn detector() -> Detector {
+        Detector::NlpRules
+    }
+
+    fn check<'a, 's>(docu: &'a Documentation, config: &Self::Config) -> Result<SuggestionSet<'s>>
+    where
+        'a: 's,
+    {
+        info!("Loading tokenizer...");
+
+        let tokenizer = config.override_tokenizer.as_ref().map_or_else(
+            || {
+                Ok(Tokenizer::from_reader(&mut &*DEFAULT_TOKENIZER_BYTES)
+                    .expect("build.rs pulls valid tokenizer description. qed"))
+            },
+            |path| -> Result<Tokenizer> {
+                let f = fs::File::open(&path)?;
+                Ok(Tokenizer::from_reader(f)?)
+            },
+        )?;
+
+        info!("Loaded tokenizer.");
+
+        info!("Loading rules..");
+
+        let rules = config.override_rules.as_ref().map_or_else(
+            || {
+                Ok(Rules::from_reader(&mut &*DEFAULT_RULES_BYTES)
+                    .expect("build.rs pulls valid tokenizer description. qed"))
```

Comment:
Seems like copy pasta:
```suggestion
                    .expect("build.rs pulls valid rules set. qed"))
```